### ISSUE TITLE
[Suggestion] Difficulty Scaling Based on Player Health

### ISSUE THREAD

**Issue by SSyl** [SilentChaos512/ScalingHealth#271]
## Version
1.12.2 (Although I'm sure it would be beneficial to all versions)

## What Should Be Added or Changed?
Add a way to track difficulty based on player health.

## Why Should This Be Done?
A few modpacks I play use Spice of Life: Carrot Edition which gives you additional hearts based on how many different foods you've eaten (which would essentially forego the additional added hearts from your mod). Eventually, you get to a point where you just have so many hearts that you're pretty much invincible, as I'm sure you know already since it's pretty much the point of this mod.

It would be nice if you could make the difficulty scale based on how many hearts a player (or nearby players) have.

For example, you could add a config option that would make it so that for every n point of HP it increases the difficulty by X.

I feel like this would fit in perfectly with all the other options in the mod and the scaling would work perfectly.

**Comment by Cyborgmas** [comment]
Yeah, can do, did not think about scaling based on hp of other mods as well. However, I dev only for 1.14+ so you wont see it in 1.12 sadly
